websites/A/Animes Vision/presence.ts

```diff
@@ -1,53 +1,109 @@
 const presence = new Presence({
 		clientId: "700405996677365842",
 	}),
-	strings = presence.getStrings({
-		playing: "general.playing",
-		paused: "general.paused",
-		browsing: "general.browsing",
-		episode: "presence.media.info.episode",
-	});
+	browsingTimestamps = Math.floor(Date.now() / 1000);
+
+enum Assets {
+	Logo = "https://i.imgur.com/CSan6Ae.png",
+	Playing = "https://i.imgur.com/KNneWuF.png",
+	Paused = "https://i.imgur.com/BtWUfrZ.png",
+}
+
+async function getStrings() {
+	return presence.getStrings(
+		{
+			playing: "general.playing",
+			paused: "general.paused",
+			episode: "general.episode",
+			browsing: "general.browsing",
+			viewHome: "general.viewHome",
+			buttonViewEpisode: "general.buttonViewEpisode",
+			buttonViewSeries: "general.buttonViewSeries",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
 
 presence.on("UpdateData", async () => {
+	const [newLang, time, showCover, showButtons] = await Promise.all([
+		presence.getSetting<string>("lang").catch(() => "en"),
+		presence.getSetting<boolean>("time"),
+		presence.getSetting<boolean>("cover"),
+		presence.getSetting<boolean>("buttons"),
+	]);
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
 	const presenceData: PresenceData = {
-			largeImageKey: "anvlogo",
-			details: (await strings).browsing,
-			startTimestamp: Math.floor(Date.now() / 1000),
+			largeImageKey: Assets.Logo,
+			details: strings.browsing,
+			startTimestamp: browsingTimestamps,
 		},
-		path = window.location.pathname;
-	if (path.endsWith("/equipe"))
-		presenceData.details = "Vendo os membros da equipe";
-
-	if (path.startsWith("/top")) presenceData.details = "Vendo o top animes";
-	else if (path.endsWith("/doramas"))
-		presenceData.details = "Vendo a lista de doramas";
-	else if (path.endsWith("/filmes"))
-		presenceData.details = "Vendo a lista de filmes";
-	else if (path.endsWith("/lancamentos"))
-		presenceData.details = "Vendo a lista de lançamentos";
-	else if (path.endsWith("/animes"))
-		presenceData.details = "Vendo a lista de animes";
-	else if (path.endsWith("/legendado")) {
-		const video: HTMLVideoElement = document.querySelector("video");
-		presenceData.details = document
-			.querySelectorAll(".active h1")[0]
-			.textContent.replace(" - Episodio ", "")
-			.replace(/[0-9]/g, "");
-		presenceData.state = (await strings).episode.replace(
-			"{0}",
-			document
-				.querySelector("#current_episode_name")
-				.textContent.match(/\d+/g)[0]
-		);
-		if (!video.paused) {
+		{ pathname, href, origin } = window.location,
+		pathArr = pathname.split("/"),
+		pathArrEnd = pathArr[pathArr.length - 1];
+
+	switch (pathArrEnd) {
+		case "":
+			presenceData.details = strings.viewHome;
+			break;
+		case "equipe":
+			presenceData.details = "Vendo os membros da equipe";
+			break;
+		case "legendado": {
```

Comment:
can you also add support for ``/dublado/`` on this part? because of the way it's currently detecting, it will only pick episodes with subs.

Reply:
Of course! I'll add it when I have the time, thanks for the suggestion

Reply:
Resolved in https://github.com/PreMiD/Presences/pull/6656/commits/33f695438fffd3e6c3e267f9a2bb363e24a25275